### ISSUE TITLE
hash_password: AttributeError accessing pepper of empty "password_config"

### ISSUE THREAD

**Issue by ivilata** [matrix-org/synapse#5315]
### Description

With Synaps 0.99.5.2, running `hash_password` yields an `AttributeError` when accessing the `pepper` key of the `password_config` section, when it is empty.

### Steps to reproduce

- Install Synapse 0.99.5.2
- Create a sample configuration file with `python -m synapse.app.homeserver --server-name example.com --config-path config.yaml --generate-config --report-stats=yes`
- Run `hash_password -c /path/to/config.yaml`

```
$ hash_password -c /path/to/config.yaml 
Traceback (most recent call last):
  File "[…]/bin/hash_password", line 56, in <module>
    password_pepper = password_config.get("pepper", password_pepper)
AttributeError: 'NoneType' object has no attribute 'get'
```

`config.yaml` contains the default `password_config` stanza, i.e.:

```yaml
password_config:
   # Uncomment to disable password login
   #
   #enabled: false

   # Uncomment and change to a secret random string for extra security.
   # DO NOT CHANGE THIS AFTER INITIAL SETUP!
   #
   #pepper: "EVEN_MORE_SECRET"
```

Patching the line causing the error to this seems to avoid the issue:

```python
        password_config = config.get("password_config", None) or {}
```

### Version information

- **Homeserver**: (own, can use fresh example setup)
- **Version**: 0.99.5.2
- **Install method**: PIP
- **Platform**: Debian GNU/Linux stable

**Comment by ivilata** [comment]
Still happening in `1.8.0`, same fix works.

**Comment by richvdh** [comment]
thanks. Could you make a PR with the patch?

**Comment by ivilata** [comment]
I'm trying to build a PR but utterly failing to figure out how to add the `changelog.d/PRnumber.type` file (as instructed by contribution guidelines) *before* creating the PR itself and thus knowing its number! I think I'm missing something, sorry! :cry:

**Comment by richvdh** [comment]
two options:

1) guess the PR number based on the current [highest issue number](https://github.com/matrix-org/synapse/issues?q=), or:
2) open the PR, then once you know the PR number, rename the changelog file to match and push it to the branch to update it.

Sorry it's a bit fiddly, but it really helps us!

**Comment by ivilata** [comment]
Thanks @richvdh! Not complaining about the procedure, but since how to do it isn't evident at all, it may be convenient to add those tricks in the doc itself.

I'll do the proper PR when I find some time. :smile:

**Comment by ivilata** [comment]
I hope #6753 is fine… in the end I just added the changelog entry in a second commit to the branch, which seemed less risky than guessing the PR number. :stuck_out_tongue:

**Comment by richvdh** [comment]
fixed by #6753. Thanks @ivilata!
